test/devise/async/model_test.rb

```diff
@@ -3,59 +3,79 @@
 module Devise
   module Async
     describe "Model" do
+
+      # As per the https://github.com/plataformatec/devise/blob/master/CHANGELOG.md#410,
+      # within transaction mail will be enqued & after commit send_devise_pending_notifications will be invoked
+
+      # 1. Within transaction i.e object creation, mails will only be accummulated 
       it "accumulates notifications to be sent after commit on Model creation" do
+        admin = create_admin
         Admin.transaction do
-          admin = create_admin
           mailers = admin.send(:devise_pending_notifications) # [:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]
-          mailers.size.must_equal 1
+          mailers.size.must_equal 0
+          Worker.expects(:enqueue).never # after_commit will not fire without save
+        end  
+        admin[:username] = "changed_username"
+        admin.send_confirmation_instructions
 
-          mailer = mailers.first
-          mailer.size.must_equal 2
-          mailer.first.must_equal :confirmation_instructions
-          mailer.last.must_be_instance_of Array
-        end
+        mailers = admin.send(:devise_pending_notifications) # [[:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]]
```

Comment:
Could you rename `mailers` to `notifications` and `mailer` to `notification`? I find the naming to be quite confusing.